tests/unit/APNSRequestTest.php

```diff
@@ -18,6 +18,23 @@ public function testRequestInstantiationFromPayload() {
 		$this->assertEquals( $message, $this->decode( $request->get_body() )->aps->alert );
 	}
 
+	public function testThatUuidCannotBeOverwrittenWithCustomUserData() {
+		$valid_uuid = $this->random_uuid();
+		$metadata   = $this->new_metadata( 'topic', $valid_uuid );
+
+		$request = APNSRequest::from_string( '', '', $metadata, $this->new_userdata_with_uuid( $this->random_uuid() ) );
+
+		$this->assertEquals( $valid_uuid, $request->get_uuid() );
+	}
```

Comment:
Nitpick. I'm not sure how random works in PHP (or anywhere else, really) but is there a chance that the two calls to `random_uuid()` here would result in the same value?

Would it be worth to do something like:

```swift
$valid_uuid = $this->random_uuid();
$different_uuid = $this->random_uuid();

$this->assertNotEqual( $valid_uuid, $different_uuid );
```

and then use `$different_uuid` to generate the `APNSRequest`?

The same observation applies to the next test.

Reply:
Interesting! I think this is rather unlikely – under the hood `random_uuid()` uses https://www.php.net/manual/en/function.random-bytes.php, which is said to be cryptographically secure, thus this shouldn't happen.